BootloaderCorePkg/Tools/CommonUtility.py

```diff
@@ -228,3 +234,43 @@ def compress (in_file, alg, out_path = '', tool_dir = ''):
 	gen_file_from_object (out_file, data)
 
 	return out_file
+
+def get_priv_key_type (priv_key, openssl_path):
+
+	print ("get_priv_key_type")
+	Output = subprocess.check_output(
+            [openssl_path, 'rsa', '-pubout', '-text', '-noout', '-in',
+             '%s' % priv_key],
+            stderr=subprocess.STDOUT).decode()
+
+	Match = re.match(r'Private-Key: (.*)\n', Output, re.M|re.I)
+	temp = re.findall(r'\d+', Match.group(1))
+	key_type = list(map(int, temp))
+
+	print ("Key Type: RSA%s" % key_type[0])
+
+	return key_type
+
+def get_sign_val (sign_type):
+	sign_val = 0
+	if sign_type == "RSA2048":
```

Comment:
use a global dictionary to make it easier

Reply:
will do.